python/ray/data/_internal/iterator/stream_split_iterator.py

```diff
@@ -137,11 +137,6 @@ def __init__(
         # Set current DataContext.
         self._data_context = dataset.context
         ray.data.DataContext._set_current(self._data_context)
-        # Automatically set locality with output to the specified location hints.
-        if locality_hints:
-            self._data_context.execution_options.locality_with_output = locality_hints
-            logger.info(f"Auto configuring locality_with_output={locality_hints}")
```

Comment:
I think here we still need something like 
```
if self._data_context.execution_options.locality_with_output is True:
    self._data_context.execution_options.locality_with_output = locality_hints
    logger.info(f"Auto configuring locality_with_output={locality_hints}")
```

Technically the above is still not correct, we should only set `locality_with_output` for the previous op. But that's an existing issue. We can just leave a TODO here for this PR.

Reply:
I think this is still needed.  
Because when `locality_with_output` is enabled, we want to place the previous map tasks on the train nodes.